business/checkers/common/multi_match_selector_checker.go

```diff
@@ -72,7 +72,12 @@ func RequestAuthenticationMultiMatchChecker(subjectType string, ra []security_v1
 func SidecarSelectorMultiMatchChecker(subjectType string, sc []networking_v1alpha3.Sidecar, workloadList models.WorkloadList) GenericMultiMatchChecker {
 	keys := []models.IstioValidationKey{}
 	selectors := make(map[int]map[string]string, len(sc))
-	for i, s := range sc {
+	i := 0
+	for _, s := range sc {
```

Comment:
Not really a problem, but sort of confusing.  The type for `selectors` is `map[int]map[string]string`.  If the keys need to be 0,1,2... then why not use an Array, which would be more clear.

Reply:
I just avoided touching the existing logic, not to cause any regression, only skipping unnecessary Sidecars.

Reply:
+1 to minimize regression risk.  It's not a problem, I just found it odd to have to worry about the int-keys being sequential, for a map.  @lucasponce, any comment?  I think this was introduced in the big client-go refactor.

Reply:
I guess that the index in the map may not be consecutive and an array may not work in those cases, but this comes after some refactoring of this checker and I'm ok with not revisiting the original design to avoid risks.

But I have an ultra-minor go-lang issue, I don't see why you'd need the "i" variable out of the "range", isn't working with the continue ?

Not a blocker :) just in case if it has a reason that I missed.

Reply:
@lucasponce
actually 'i' index is required to be as a sequence, used in mapping indexes in keys and sequences, otherwise by the same index the key will be different one, that is why I did not want to touch the logic :)

Reply:
...and this is exactly why I pointed out this confusing area :-)